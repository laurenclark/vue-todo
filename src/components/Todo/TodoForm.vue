<template>
    <label for="">Add Todo:</label>
    <input type="text" />
</template>

<script>
export default {
    name: "TodoForm"
};
</script>

<style>
</style>