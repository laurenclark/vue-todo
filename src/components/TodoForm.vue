<script>
export default {
    name: "TodoForm"
};
</script>

<template>
    <label for="">Add Todo:</label>
    <input type="text" />
</template>

<style>
</style>