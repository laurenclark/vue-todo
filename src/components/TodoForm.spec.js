import TodoForm from "./TodoForm.vue";
import { mount } from "@vue/test-utils";

describe("Is the TodoForm Component", () => {
    it("Exports a valid component", () => {
        expect(TodoForm).toBeAComponent();
    });
    it("Renders correctly", () => {
        const wrapper = mount(TodoForm);
        expect(wrapper.element).toMatchSnapshot();
    });
});

describe("Blank slate of the form", () => {
    const wrapper = mount(TodoForm);
    // Get/Find: Get is similar to find, but get throws instead of returning a ErrorWrapper.
    // As a rule of thumb, always use [get] except when you are asserting something doesn't exist.
    // In that case use [find].
    const textInput = wrapper.get("input[text]");
    it("Has a text input", () => {
        expect(wrapper).toMatchSnapshot();
        expect(textInput.exists()).toBe(true);
    });
    it("Has placeholder text in the text input", () => {
        const expected = "Wash dishes";
        expect(textInput.html()).toContain(expected);
    });
    // it("Has a date picker input", () => {
    //     expect(true).toBe(false);
    // });
    // it("Has a button labelled 'submit'", () => {
    //     expect(true).toBe(false);
    // });
});

// describe("Creates a Todo Object", () => {
//     expect(true).toBe(false);
// });
