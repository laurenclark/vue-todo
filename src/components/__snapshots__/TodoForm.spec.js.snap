// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blank slate of the form Has a text input 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      
      <input
        placeholder="Wash the dishes"
        text=""
        type="text"
      />
      <input
        date="Thu Jan 06 2022 01:45:22 GMT+0000 (Greenwich Mean Time)"
        type="date"
      />
      <button>
        Add Todo
      </button>
      
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {},
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {},
      "refs": Object {
        "VTU_COMPONENT": Object {
          "hasOwnProperty": [Function],
          "todoDate": 2022-01-06T01:45:22.560Z,
          "todoText": "",
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "hasOwnProperty": 4,
            "isVueInstance": 0,
            "toJSON": 0,
            "todoDate": 2,
            "todoText": 2,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "hasOwnProperty": [Function],
            "todoDate": 2022-01-06T01:45:22.560Z,
            "todoText": "",
          },
          "da": null,
          "data": Object {
            "todoDate": 2022-01-06T01:45:22.560Z,
            "todoText": "",
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {},
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {},
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "hasOwnProperty": [Function],
            "todoDate": 2022-01-06T01:45:22.560Z,
            "todoText": "",
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": ,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "text",
                ],
                "el": <input
                  placeholder="Wash the dishes"
                  text=""
                  type="text"
                />,
                "key": null,
                "patchFlag": 8,
                "props": Object {
                  "placeholder": "Wash the dishes",
                  "text": "",
                  "type": "text",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "date",
                ],
                "el": <input
                  date="Thu Jan 06 2022 01:45:22 GMT+0000 (Greenwich Mean Time)"
                  type="date"
                />,
                "key": null,
                "patchFlag": 8,
                "props": Object {
                  "date": 2022-01-06T01:45:22.560Z,
                  "type": "date",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Add Todo",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <button>
                  Add Todo
                </button>,
                "key": null,
                "patchFlag": -1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "button",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "text",
                ],
                "el": <input
                  placeholder="Wash the dishes"
                  text=""
                  type="text"
                />,
                "key": null,
                "patchFlag": 8,
                "props": Object {
                  "placeholder": "Wash the dishes",
                  "text": "",
                  "type": "text",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "date",
                ],
                "el": <input
                  date="Thu Jan 06 2022 01:45:22 GMT+0000 (Greenwich Mean Time)"
                  type="date"
                />,
                "key": null,
                "patchFlag": 8,
                "props": Object {
                  "date": 2022-01-06T01:45:22.560Z,
                  "type": "date",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
            ],
            "dynamicProps": null,
            "el": ,
            "key": null,
            "patchFlag": 64,
            "props": null,
            "ref": null,
            "scopeId": null,
            "shapeFlag": 16,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": Symbol(Fragment),
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "data": [Function],
            "name": "TodoForm",
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "data": [Function],
          "name": "TodoForm",
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.26",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "hasOwnProperty": [Function],
    "todoDate": 2022-01-06T01:45:22.560Z,
    "todoText": "",
  },
  "isDisabled": [Function],
  "rootVM": Object {},
  "wrapperElement": ,
}
`;

exports[`Is the TodoForm Component Renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <input
    placeholder="Wash the dishes"
    text=""
    type="text"
  />
  <input
    date="Thu Jan 06 2022 01:45:22 GMT+0000 (Greenwich Mean Time)"
    type="date"
  />
  <button>
    Add Todo
  </button>
  
</div>
`;
